packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -847,9 +847,11 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
 ///  * [WidgetsBinding.handleBeginFrame], which pumps the widget pipeline to
 ///    ensure the widget, element, and render trees are all built.
 void runApp(Widget app) {
-  WidgetsFlutterBinding.ensureInitialized()
-    ..attachRootWidget(app)
-    ..scheduleWarmUpFrame();
+  final WidgetsBinding binding = WidgetsFlutterBinding.ensureInitialized();
+  Timer.run(() {
```

Comment:
I might as well create an api to just schedule attachRootWidget and flutter test can override this api. This might make  it more clean, but I am not sure if it worth to expose an api for that